File: VB/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraReports.UI

' ...
Namespace docCustomDataItemsNames

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim report As XtraReport1 = New XtraReport1()
            report.DataSource = New MySqlDataSource(CType(report.DataSource, SqlDataSource))
            Dim designTool As ReportDesignTool = New ReportDesignTool(report)
            designTool.ShowDesignerDialog()
        End Sub
    End Class
End Namespace

File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraReports.UI;
// ...

namespace docCustomDataItemsNames {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            report.DataSource = new MySqlDataSource((SqlDataSource)report.DataSource);
            ReportDesignTool designTool = new ReportDesignTool(report);
            designTool.ShowDesignerDialog();
        }
    }
}